baremetal_vm/README.md

```diff
@@ -28,74 +28,102 @@ This script creates a VM to emulate a bare-metal host, intended for developer te
+the DHCP range for the libvirt network, e.g using `virsh net-edit` adjust the `start` similar to below, which allows us to reserve ips under `.20`
+for static assignment:
 
 ```
-> ./get_ip.sh
-192.168.122.116
-
-> cat > /tmp/dns.xml << EOF
-<host ip='192.168.122.116'>
-  <hostname> boot.ironic.suse.baremetal </hostname>
-  <hostname> api.ironic.suse.baremetal </hostname>
-  <hostname> inspector.ironic.suse.baremetal </hostname>
-  <hostname> media.suse.baremetal </hostname>
-</host>
-EOF
-> virsh net-update default add --section dns-host --xml dns.xml --live
+  <ip address='192.168.122.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.122.20' end='192.168.122.254'/>
+    </dhcp>
+  </ip>
 ```
 
+Note that after editing the network you must `virsh net-destroy` then `virsh net-start` for the changes to take effect (which will disconnect
+any running VMs so they will also require a destroy/start).
+
 #### Metal<sup>3</sup> chart installation
 
 The metal3 chart can be installed with a minimal configuration via the following steps:
 
 Note this assumes that kubectl is configured e.g via the `KUBECONFIG` variable to connect to the management cluster created in the previous step.
 
-First enable the charts repo and create a values file which disables the provisioning network:
+First deploy cert-manager and MetalLB - in this example we're using `192.168.122.10` as the reserved IP for access to Ironic services:
+
 
 ```
+helm repo add jetstack https://charts.jetstack.io
 helm repo add suse-edge https://suse-edge.github.io/charts
 helm repo update
 
-cat > disable_provnet.yaml <<EOF
-global:
-  enable_dnsmasq: false
-  enable_pxe_boot: false
-  provisioningInterface: ""
-  provisioningIP: ""
-  enable_metal3_media_server: false
+helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --version v1.11.1 --set installCRDs=true
+helm install metallb suse-edge/metallb --namespace metallb-system --create-namespace
+
+export STATIC_IRONIC_IP=192.168.123.10
+cat <<-EOF | kubectl apply -f -
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: ironic-ip-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - ${STATIC_IRONIC_IP}/32
+  serviceAllocation:
+    priority: 100
+    serviceSelectors:
+    - matchExpressions:
+      - {key: app.kubernetes.io/name, operator: In, values: [metal3-ironic]}
+EOF
+
+cat <<-EOF | kubectl apply -f -
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: ironic-ip-pool-l2-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+  - ironic-ip-pool
 EOF
 ```
 
-Then install the chart from the same directory, using the values file:
+Then we deploy Metal<sup>3</sup>
 
 ```
-helm install metal3 suse-edge/metal3 -f ./disable_provnet.yaml
+cat > values.yaml <<EOF
+global:
+  ironicIP: ${STATIC_IRONIC_IP}
+EOF
+
+helm install metal3 suse-edge/metal3 --namespace metal3-system --create-namespace -f values.yaml
```

Comment:
Personal preference but if it is just a single parameter I prefer:
```
helm install metal3 suse-edge/metal3 --namespace metal3-system --create-namespace --set global.ironicIP=${STATIC_IRONIC_IP}
```

Reply:
Ack yeah I thought the same - previously we passed multiple values but now it's just one it can be done via `--set` - will update!

Reply:
Done!